Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -178,15 +177,53 @@ public async Task<ContestDetailsServiceModel> GetContestDetails(int id)
             registerModel.RequirePassword = ShouldRequirePassword(contest!, participant!, isOfficial);
             registerModel.ParticipantId = participant?.Id;
             registerModel.IsRegisteredSuccessfully = participant != null;
-            registerModel.ShouldConfirmParticipation = contest!.IsOnlineExam &&
-                                                       isOfficial &&
-                                                       participant == null &&
-                                                       !userIsAdminOrLecturerInContest;
+            registerModel.ShouldConfirmParticipation = ShouldConfirmParticipation(participant, isOfficial, contest!.IsOnlineExam, userIsAdminOrLecturerInContest);
+
+            return registerModel;
+        }
+
+        public async Task<bool> ValidateUserRegistrationForContest(
+            int id,
+            string? password,
+            bool? hasConfirmedParticipation,
+            bool isOfficial)
+        {
+            var contest = this.contestsData
+                .GetByIdQuery(id)
+                .Include(c => c.Category)
+                .Include(c => c.ProblemGroups)
+                .ThenInclude(pg => pg.Problems)
+                .FirstOrDefault();
+
+            var user = this.userProviderService.GetCurrentUser();
+            var userProfile = await this.usersBusinessService.GetUserProfileById(user.Id);
+
+            var validationResult = this.contestParticipationValidationService.GetValidationResult((
+                contest,
+                id,
+                user,
+                isOfficial)!);
+
+            if (!validationResult.IsValid)
+            {
+                throw new BusinessServiceException(validationResult.Message);
+            }
+
+            var participant = await this.participantsData
+                .GetByContestByUserAndByIsOfficial(
+                    id,
+                    userProfile!.Id,
+                    isOfficial);
+
+            var userIsAdminOrLecturerInContest = await this.lecturersInContestsBusiness.IsCurrentUserAdminOrLecturerInContest(contest?.Id);
+            bool shouldRequirePassword = ShouldRequirePassword(contest!, participant, isOfficial);
+            bool shouldConfirmParticipation =
+                ShouldConfirmParticipation(participant, isOfficial, contest!.IsOnlineExam, userIsAdminOrLecturerInContest);
```

Comment:
Can we reuse the method where this information is retrieved in `RegisterUserForContestServiceModel`

Reply:
We do, this change does this so both methods use ShouldRequirePassword ShouldConfirmParticipation